src/main/kotlin/no/nav/syfo/identhendelse/kafka/IdenthendelseConsumerService.kt

```diff
@@ -12,30 +14,35 @@ import kotlin.time.Duration.Companion.seconds
 
 class IdenthendelseConsumerService(
     private val identhendelseService: IdenthendelseService,
-) {
-    suspend fun pollAndProcessRecords(kafkaConsumer: KafkaConsumer<String, GenericRecord>) {
-        try {
-            val records = kafkaConsumer.poll(Duration.ofSeconds(POLL_DURATION_SECONDS))
-            if (records.count() > 0) {
-                records.forEach { record ->
-                    if (record.value() != null) {
-                        identhendelseService.handleIdenthendelse(record.value().toKafkaIdenthendelseDTO())
-                    } else {
-                        log.warn("Identhendelse: Value of ConsumerRecord from topic $PDL_AKTOR_TOPIC is null, probably due to a tombstone. Contact the owner of the topic if an error is suspected")
-                        COUNT_KAFKA_CONSUMER_PDL_AKTOR_TOMBSTONE.increment()
+) : KafkaConsumerService<GenericRecord> {
```

Comment:
Hva for noe spennende får vi av bruke KafkaConsumerService her?

Reply:
Er nok kun strukturelt og for ryddigheten + man får litt mindre duplisert kode for å sette opp en kafka-consumer siden `launchKafkaTask` tar seg av å launche backgroundtask, sette opp kafka-consumer og subscribe til topic.